components/brave_wallet/browser/keyring_service.cc

```diff
@@ -2520,6 +2554,56 @@ void KeyringService::HasPendingUnlockRequest(
   std::move(callback).Run(HasPendingUnlockRequest());
 }
 
+absl::optional<std::vector<std::pair<std::string, mojom::ZCashKeyIdPtr>>>
+KeyringService::GetZCashAddresses(const mojom::AccountId& account_id) {
+  CHECK(IsZCashAccount(account_id));
+
+  auto* zcash_keyring = GetZCashKeyringById(account_id.keyring_id);
+  if (!zcash_keyring) {
+    return absl::nullopt;
+  }
+
+  // TODO(cypt4): store used addresses indexes in prefs.
+
+  // TODO(cypt4): temporarily just return first 5 recieve and 5 change
+  // addresses.
+  std::vector<std::pair<std::string, mojom::ZCashKeyIdPtr>> addresses;
```

Comment:
I don't think we should assume that the first will always be useful here. Should we check that the addresses are unused via the RPC instead? E.g. someone recovers an already used seed phrase to our wallet then we wouldn't want to reuse these addresses is what I'm thinking.

Reply:
Addresses discovery will be implemented in the future